File: src/main/java/com/morak/performancetracker/aop/rest/RestMonitor.java
package com.morak.performancetracker.aop.rest;

import com.morak.performancetracker.Monitor;
import org.springframework.stereotype.Component;

@Component
public class RestMonitor implements Monitor {

    private String uri = null;
    private long startTime = 0;
    private long elapsedTime = 0;

    public void start(String uri) {
        this.uri = uri;
        this.startTime = System.nanoTime();
    }

    public void end() {
        this.elapsedTime = this.startTime - elapsedTime;
    }

    @Override
    public void clear() {
        this.uri = null;
        this.startTime = 0;
        this.elapsedTime = 0;
    }

    @Override
    public String getSignature() {
        return "REST";
    }

    public String getUri() {
        return uri;
    }

    public long getStartTime() {
        return startTime;
    }

    public long getElapsedTime() {
        return elapsedTime;
    }
}

File: src/main/kotlin/com/morak/performancetracker/aop/rest/RestMonitor.kt
package com.morak.performancetracker.aop.rest

import com.morak.performancetracker.Monitor
import org.springframework.stereotype.Component

@Component
class RestMonitor : Monitor {
    var uri: String? = null
    var startTime: Long = 0
    var elapsedTime: Long = 0

    fun start(uri: String?) {
        this.uri = uri
        startTime = System.nanoTime()
    }

    fun end() {
        elapsedTime = System.nanoTime() - startTime
    }

    override fun clear() {
        uri = null
        startTime = 0
        elapsedTime = 0
    }

    override val signature: String
        get() = "REST"
}
